solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/Feature.java

```diff
@@ -141,14 +146,29 @@ public void setId(int id) {
     this.id = id;
   }
 
+  public static Feature fromMap(SolrResourceLoader solrResourceLoader,
+      Map<String,Object> featureMap) {
+    final String type =
+        (String) featureMap.get(CLASS_KEY);
+
+    final String name =
+        (String) featureMap.get(NAME_KEY);
+
+    @SuppressWarnings("unchecked")
+    final Map<String,Object> params =
+    (Map<String,Object>) featureMap.get(PARAMS_KEY);
+
+    return Feature.getInstance(solrResourceLoader,
+        type, name, params);
+  }
+
   protected abstract LinkedHashMap<String,Object> paramsToMap();
 
-  public LinkedHashMap<String,Object> toMap(String storeName) {
-    final LinkedHashMap<String,Object> o = new LinkedHashMap<>(4, 1.0f);
-    o.put((String)CommonLTRParams.FEATURE_NAME, name);
-    o.put((String)CommonLTRParams.FEATURE_CLASS, getClass().getCanonicalName());
-    o.put((String)CommonLTRParams.FEATURE_STORE, storeName);
-    o.put((String)CommonLTRParams.FEATURE_PARAMS, paramsToMap());
+  public LinkedHashMap<String,Object> toMap() {
```

Comment:
The ManagedFeatureStore is the only class that actually cares about the JSON/map representation of the feature, so I'm thinking that it should own toMap and fromMap as static functions.  It's the only one that calls them in addFeature() and featuresAsManagedResources().
These will be 2 less functions that clutter the parent Feature class which many people will be looking at as they subclass it for their specialized features.